starlette/datastructures.py

```diff
@@ -428,28 +427,24 @@ class UploadFile:
     An uploaded file included as part of the request data.
     """
 
-    spool_max_size = 1024 * 1024
-    file: typing.BinaryIO
-    headers: "Headers"
-
     def __init__(
         self,
-        filename: str,
-        file: typing.Optional[typing.BinaryIO] = None,
-        content_type: str = "",
+        file: typing.BinaryIO,
         *,
+        filename: typing.Optional[str] = None,
         headers: "typing.Optional[Headers]" = None,
     ) -> None:
         self.filename = filename
-        self.content_type = content_type
-        if file is None:
-            self.file = tempfile.SpooledTemporaryFile(max_size=self.spool_max_size)  # type: ignore[assignment]  # noqa: E501
-        else:
-            self.file = file
+        self.file = file
         self.headers = headers or Headers()
 
+    @property
+    def content_type(self) -> typing.Optional[str]:
+        return self.headers.get("content-type", None)
```

Comment:
Should we maintain the same behavior as before here?
```suggestion
    def content_type(self) -> str:
        return self.headers.get("content-type", "")
```

Reply:
Ok. But I believe the original behavior doesn’t make much sense.

Reply:
Maybe we can put this on the list of breaking changes for 1.0?

Reply:
We can break it here then... :+1: